### ISSUE TITLE
IE10 "clear" functionality compatibility

### ISSUE THREAD

**Issue by Macharius92** [RobinHerbots/Inputmask#304]
Hi @RobinHerbots 

Can you please provide a correction for the "clear" functionality in IE10.

I set a mask on my input.
I enter a valid value and click in other input, it's ok.
I'm now clearing the input with the little cross at the right of the field (only IE10) and clicking in an other input, the masked input is empty.
I'm move my mouse over the masked input, it restores the old value.

i suggest you to look towards the "input" event or binding the clearing on the "mouseup" ;)

**Comment by RobinHerbots** [comment]
@Macharius92 , 

I tested the scenario with IE10 and do not have this behavior.

Can you make a jsfiddle for it.

**Comment by Macharius92** [comment]
@RobinHerbots 

You're right it works fine with IE10 in IE10 mode... I was in another IE mode (tested with IE9 in the browser mode of developer tools). And then you have the "clear" cross in the input but it doesn't work as IE10 ;)

**Comment by RobinHerbots** [comment]
@Macharius92, 

In whatever mode you put IE10 it's shows his clearingcross.   I don't know that the ms-clear is feature detectable.

For the moment the inputmask detects IE10 and act accordingly to the input event.

**Comment by sompylasar** [comment]
I resolved this issue by hiding the cross via CSS for an inputmask field.

**Comment by RobinHerbots** [comment]
The functionality is working for IE10, but is via browser detection and not feature detection.

**Comment by peterkeller** [comment]
The workaround seems only to be working for IE10 but not for IE11, IE9 and IE8. Is there any possibility that this will be fixed for these versions (for me, IE8 is crucial as our customer still uses this version...)?

**Comment by RobinHerbots** [comment]
@peterkeller, 

I will fix it for ie11, but is their a clear (cross) functionality in ie8 and ie9 ?  I mean in the real ie8?

**Comment by peterkeller** [comment]
I guess you are right: The "cross" definitely appears in IE11. However, I didn't test it with the "real" IE8 and IE9 but only with the emulations available in IE11. Thanks!

**Comment by sompylasar** [comment]
The "clear" button appears since IE10. The real IE9 and IE8 do not contain
this button, but the emulated IE8 and IE9 in IE10 and IE11 do.
https://coderwall.com/p/ex-jjq

**Comment by RobinHerbots** [comment]
@peterkeller , 

The latest in branch 2.x should support IE11.  Can you have a test?

best regards, 
Robin

**Comment by RobinHerbots** [comment]
@peterkeller , 

It is not working.  I will try to detect IE11 with 

``` javascript
 var tridentVersion = 
     typeof ScriptEngineMajorVersion === "function" ?
         ScriptEngineMajorVersion() : undefined
```

Will add that tomorrow for test

**Comment by RobinHerbots** [comment]
@peterkeller , 

Ok the latest version should detect ie11 & IE10 properly.

Can you have a try

**Comment by peterkeller** [comment]
Thank you very much for the fix. The "cross" functionality seems to work in IE11. As I don't have an original IE10, I cannot test the fix for this version. I used the scripts at https://github.com/RobinHerbots/jquery.inputmask/blob/2.6 for my tests.

However I found another bug that may be related to this issue: 
- I enter a text
- I set the cursor in the middle of the text and delete the text with back-delete from the cursor to the beginning of the text field
- When I reach the beginning of the text field, the whole text is deleted of the right (!) side of the cursor so that the whole text  input field is blank. After that, it is not possible to enter any text in the field.
